frontends/web/src/routes/bitboxbase/bitboxbase.tsx

```diff
@@ -166,19 +171,32 @@ class BitBoxBase extends Component<Props, State> {
 
     private onStatusChanged = () => {
         apiGet(this.apiPrefix() + '/status').then(({status}) => {
-            if (!this.state.showWizard && ['connected', 'unpaired', 'pairingFailed', 'bitcoinPre'].includes(status)) {
+            if (!this.state.showWizard && ['connected', 'unpaired', 'pairingFailed', 'passwordNotSet', 'bitcoinPre'].includes(status)) {
                 this.setState({ showWizard: true });
             }
             this.setState({
                 status,
             });
-            // Dummy check for automatic paired response from base until we have an rpc response and event from the base
-            if (this.state.status === 'bitcoinPre') {
-                this.setState({activeStep: ActiveStep.SetPassword});
-            }
-            if (this.state.status === 'initialized') {
-                this.onNewMiddlewareInfo();
-                this.onNewVerificationProgress();
+            // check if the base middleware password has been set yet
+            switch (this.state.status) {
+                case 'passwordNotSet':
+                    this.setState({activeStep: ActiveStep.SetPassword});
+                    break;
+                case 'bitcoinPre':
+                    this.setState({activeStep: ActiveStep.ChooseSetup});
+                    break;
+                case 'locked':
+                    this.setState({ showWizard: false });
+                    this.setState({ locked: true });
+                    break;
+                case 'initialized':
+                    this.setState({ locked: false });
+                    this.setState({ showWizard: false });
```

Comment:
you could use 1 this.setState call, i.e.
```js
this.setState({
  locked: false,
  showWizard: false 
})
```

Reply:
thanks